file/validate.go

```diff
@@ -29,5 +87,16 @@ func validate(content []byte) error {
 		err := fmt.Errorf(desc.String())
 		errs.Errors = append(errs.Errors, err)
 	}
+
+	var kongdefaults *KongDefaults
+	if err := json.Unmarshal(content, kongdefaults); err != nil {
+		return fmt.Errorf("unmarshaling file into KongDefaults: %w", err)
+	}
+	if kongdefaults != nil {
+		if err := checkDefaults(*kongdefaults); err != nil {
+			return fmt.Errorf("fields are not allowed to specify in defaults")
```

Comment:
We do want to include the downstream error text here though:
```suggestion
			return fmt.Errorf("default values are not allowed for these fields: %w", err)
```
Also some rephrasing--the transitive "to specify" with no object in the original was confusing.